lib/hamt.js

```diff
@@ -528,31 +538,46 @@ Map.prototype.isEmpty = function() {
 /**
     Alter the value stored for `key` in `map` using function `f` using
     custom hash.
-    
+
     `f` is invoked with the current value for `k` if it exists,
     or no arguments if no such value exists. `modify` will always either
     update or insert a value into the map.
-    
+
     Returns a map with the modified value. Does not alter `map`.
 */
-const modifyHash = hamt.modifyHash = (f, hash, key, map) =>
-    new Map(map.root._modify(0, f, hash, key));
+const modifyHash = hamt.modifyHash = function(f, hash, key, defaultValue, map) {
+    if(arguments.length <= 4) {
+        map = defaultValue;
+        defaultValue = nothing;
+    }
+
+    f = defaultVal(f, defaultValue);
+
+    return new Map(map.root._modify(0, f, hash, key));
+};
 
-Map.prototype.modifyHash = function(hash, key, f) {
-    return modifyHash(f, hash, key, this);
+Map.prototype.modifyHash = function(hash, key, f, defaultValue) {
+    defaultValue = arguments.length <= 3 ? nothing : defaultValue;
```

Comment:
To remove the need to additional logic involving `nothing`, try duplicating the calling code instead of adding more logic further down:

```js
Map.prototype.modifyHash = function(hash, key, f, defaultValue) {
    return arguments.length <= 3 ? modifyHash(f, hash, key, this) : modifyHash(f, hash, key, defaultValue, this);
```